core/commands/storage.go

```diff
@@ -302,139 +216,389 @@ Use status command to check for completion:
 				return err
 			}
 		}
-		// start new session
-		// TODO: Genereate session ID on new
-		sm := storage.GlobalSession
-		ssID, err := storage.NewSessionID()
-		// start new session
+
+		output, err := openSession(&paramsForOpenSession{
+			req:         req,
+			n:           n,
+			cfg:         cfg,
+			ctx:         req.Context,
+			rootHash:    rootHash,
+			runMode:     runMode,
+			shardHashes: shardHashes,
+			blacklist:   set.New(),
+			shardSize:   shardSize,
+		})
 		if err != nil {
 			return err
 		}
-		ss := sm.GetOrDefault(ssID, n.Identity)
-		ss.SetFileHash(rootHash)
-		ss.SetRunMode(runMode)
-		initSessionStatus := storage.InitStatus
-		if ss.IsOffSignRunmode() {
-			ss.OfflineCB = new(storage.OfflineControlBlock)
-			initSessionStatus = storage.UninitializedStatus
-			initOfflineSignChannels(ss)
+
+		go retryMonitor(storage.NewGoContext(req.Context), api, output.ss, n, output.ssID, output.testFlag,
+			runMode, renterPid.Pretty(), output.customizedSchedule, output.period)
+
+		seRes := &UploadRes{
+			ID: output.ssID,
 		}
-		ss.SetStatus(initSessionStatus)
-		go controlSessionTimeout(ss, storage.StdSessionStateFlow[0:], initSessionStatus)
+		return res.Emit(seRes)
+	},
+	Type: UploadRes{},
+}
 
-		// get hosts/peers
-		var peers []string
-		// init retry queue
-		retryQueue := storage.NewRetryQueue(int64(len(peers)))
-		// set price limit, the price is default when host doesn't provide price
-		price, found := req.Options[uploadPriceOptionName].(int64)
-		if found && price < HostPriceLowBoundary {
-			return fmt.Errorf("price is smaller than minimum setting price")
+var storageUploadRepairCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Repair specific shards of a file.",
+		ShortDescription: `
+This command repairs the given shards of a file.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("file-hash", true, false, "Hash of file to upload."),
+		cmds.StringArg("repair-shards", false, false, "Shard hashes to repair."),
+		cmds.StringArg("renter-pid", false, false, "Original renter peer ID."),
+		cmds.StringArg("blacklist", false, false, "Blacklist of hosts during upload."),
```

Comment:
@laipogo Should the first 3 arguments be required arguments now? since repair is a separate command?